File: Jump game/sol.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let size = nums.length;
    let val = 0;
    for(let i=0;i<size;i++)
    {
        if(i>val)
        {
            return false;
        }
        val = Math.max(val, i+nums[i]);
    }
    return true;
};

File: Jump game/sol.java
class Solution {
    public boolean canJump(int[] nums) {
        int size = nums.length;
        int val = 0;
        for(int i=0;i<size;i++)
        {
            if(i>val)
            {
                return false;
            }
            val = Math.max(val, i+nums[i]);
        }
        return true;
    }
}
